testng-core/src/test/java/test/testng1232/TestListenerFor1232.java

```diff
@@ -5,13 +5,17 @@
 import java.util.concurrent.atomic.AtomicInteger;
 import org.testng.*;
 import org.testng.collections.Maps;
+import org.testng.internal.AutoCloseableLock;
 import org.testng.xml.XmlSuite;
 
 public class TestListenerFor1232 extends ListenerTemplate {
   static Map<CounterTypes, AtomicInteger> counters = Maps.newHashMap();
+  private static final AutoCloseableLock lock = new AutoCloseableLock();
```

Comment:
```suggestion
  private final AutoCloseableLock lock = new AutoCloseableLock();
```

Previously this lock was instance-based, so `static` should not be here

Reply:
No. I am just replacing the `static synchronized resetCounters()` method with this static lock.